services/web/server/tests/unit/isolated/test_exporter_xlsx.py

```diff
@@ -0,0 +1,484 @@
+# pylint: disable=redefined-outer-name
+from typing import Any, Dict
+from pathlib import Path
+import random
+import string
+
+
+import openpyxl
+from openpyxl import Workbook
+import pytest
+
+
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.submission import (
+    SubmissionXLSXDocument,
+    SubmissionDocumentParams,
+)
+
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.dataset_description import (
+    DatasetDescriptionXLSXDocument,
+    DatasetDescriptionParams,
+    ContributorEntryModel,
+    DoiEntryModel,
+    LinkEntryModel,
+)
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.code_description import (
+    CodeDescriptionXLSXDocument,
+    CodeDescriptionParams,
+    CodeDescriptionModel,
+    InputsEntryModel,
+    OutputsEntryModel,
+    RRIDEntry,
+)
+from simcore_service_webserver.exporter.formatters.sds.xlsx.templates.directory_manifest import (
+    DirectoryManifestXLSXDocument,
+    DirectoryManifestParams,
+    FileEntryModel,
+)
+
+
+MAX_ENTRIES_IN_ARRAYS = 10
+
+
+# fixtures below
+
+
+@pytest.fixture
+def temp_dir(tmpdir) -> Path:
+    # cast to Path object
+    yield Path(tmpdir)
+
+
+# utils below
+
+
+def get_workbook(xls_path: Path) -> Workbook:
+    return openpyxl.load_workbook(xls_path)
+
+
+def assert_expected_layout(
+    workbook: Workbook, expected_layout: Dict[str, Dict[str, Any]]
+) -> bool:
+    for sheet_name in expected_layout:
+        sheet = workbook[sheet_name]
+        sheet_expected_values = expected_layout[sheet_name]
+        for address, value in sheet_expected_values.items():
+            sheet_value = sheet[address].value
+            assert (
+                sheet_value == value
+            ), f"'{address} value' \"{sheet_value}\" != 'expected value' \"{value}\""
+
+    return True
+
+
+RANDOM_CHARS = (
+    string.ascii_lowercase
+    + string.ascii_uppercase
+    + string.ascii_letters
+    + string.digits
+)
+
+
+def _random_word() -> str:
+    return "".join(random.choice(RANDOM_CHARS) for i in range(random.randrange(5, 20)))
+
+
+def random_text(prefix: str = "") -> str:
+    prefix_str = f"{prefix}__" if prefix else ""
```

Comment:
please for all fake data USE https://faker.readthedocs.io/en/master/
and if you customize something, add it under ``packages/pytest-simcore/src/pytest_simcore/helpers`` so we can all reuse it!

Reply:
totally forgot about faker :\